waiter/src/waiter/util/utils.clj

```diff
@@ -527,6 +528,36 @@
                   (sleep delay-ms)
                   (recur (inc num-tries) (* delay-ms delay-multiplier))))))))
 
+(defn async-retry-strategy
+  "Return a async retry function using the specified retry config.
+   The returned function accepts a no-args async function to be executed.
+
+   `delay-multiplier` each previous delay is multiplied by delay-multiplier to generate the next delay.
+   `initial-delay-ms` the initial delay for the first retry.
+   `max-delay-ms` the delay cap for exponential backoff delay.
+   `max-retries`  limit the number of retries.
+   "
+  [{:keys [delay-multiplier initial-delay-ms max-delay-ms max-retries]
+    :or {delay-multiplier 1.0
+         initial-delay-ms 100
+         max-delay-ms 300000 ; 300k millis = 5 minutes
+         max-retries 10}}]
+  (fn [body-function]
```

Comment:
Nitpick: This function forces the body to have to launch its own `async/go` block and do error handling to return an `Exception` during failure. We should mention this clearly in the documentation.